### ISSUE TITLE
Support for default httpRequest to use URL defined in the config for transaction functions

### ISSUE THREAD

**Issue by garrettmac** [okta/okta-auth-js#711]
so the sdk creates an instance with a config but we use our own domain that then points to the `https://{yourOktaDomain}`

So for example

```
var OktaAuth = require('@okta/okta-auth-js')
var config = {
  issuer: "https://{mycustomdomain}.com" // normally the okta domain
};
var authClient = new OktaAuth(config);
```
this authClient has almost all of the functions for us, pointing to "https://{mycustomdomain}.com"
`authClient.login()`
`authClient.logout()`
`authClient.verifyRecoveryToken()`
ect.

however resetPassword does not sit on the authClient instance. so their is no `authClient.resetPassword()`
instead its from the transaction from verifyRecoveryToken. so it looks something like. so to call it we'd need to do something like
```js
authClient.verifyRecoveryToken({
  recoveryToken: '00xdqXOE5qDZX8-PBR1bYv8AESqIFinDy3yul01tyh'
})
.then(function(transaction) {
  transaction.resetPassword()
})
```
but the transaction api calls call `https://{yourOktaDomain}` and not "https://{mycustomdomain}.com". I want a config level change that will support this.

Also see the PR. https://github.com/okta/okta-auth-js/pull/710 it might need optional chaining or another conditional check but can you please support

**Comment by shuowu** [comment]
@garrettmac The `resetPassword` action depends on the `recoveryToken`, it needs to be called after retrieving the token first. Please refer to the [transaction status](https://github.com/okta/okta-auth-js#transactionstatus) for details.

**Comment by garrettmac** [comment]
@shuowu yep so im starting the transaction via verifyRecoveryToken. that aside when calling transaction.resetPassword() its going to https://{yourOktaDomain} and not "https://{mycustomdomain}.com"

**Comment by shuowu** [comment]
@garrettmac Sorry missed that part in your description. Right, it looks like a valid case.
Internal Ref: OKTA-390664

**Comment by garrettmac** [comment]
i dont think this code fixes it, and kinda hard to see where the fix needs to be, im having a hard time seeing where youre transactions functions are....

**Comment by garrettmac** [comment]
i need this pretty urgently. can you point me where the change needs to be and i can install via a forked branch for now?

**Comment by garrettmac** [comment]
Can I have someone look at my PR @shuowu. We'd like to have this patch ASAP https://github.com/okta/okta-auth-js/pull/710#discussion_r625885941
